onedal/tests/utils/_device_selection.py

```diff
@@ -73,26 +73,14 @@ def is_dpctl_device_available(targets):
     return False
 
 
-def device_type_to_str(queue):
-    if queue is None:
-        return "cpu"
-
-    if dpctl_available:
-        if queue.sycl_device.is_cpu:
-            return "cpu"
-        if queue.sycl_device.is_gpu:
-            return "gpu"
-    return "unknown"
-
-
 def pass_if_not_implemented_for_gpu(reason=""):
     assert reason
 
     def decorator(test):
         @functools.wraps(test)
         def wrapper(queue, *args, **kwargs):
             if queue is not None and queue.sycl_device.is_gpu:
-                with pytest.raises(RuntimeError, match="is not implemented for GPU"):
+                with pytest.raises(RuntimeError, match=reason):
```

Comment:
I can't understand why this has been hardcoded previously. We even do `assert reason`.
In the `main` version, the provided RegEx to the decorator is completely ignored and we only compare against `is not implemented for GPU`. The changed version fixes this, which is why the RegExes in the test cases had to be updated.